webrender/src/texture_cache.rs

```diff
@@ -746,13 +778,28 @@ impl TextureCache {
     pub fn update(&mut self,
                   image_id: TextureCacheItemId,
                   descriptor: ImageDescriptor,
+                  filter: TextureFilter,
                   data: ImageData,
                   dirty_rect: Option<DeviceUintRect>) {
-        let existing_item = self.items.get(image_id);
+        let existing_item = self.items.get(image_id).clone();
+
+        if existing_item.allocated_rect.size.width != descriptor.width ||
+           existing_item.allocated_rect.size.height != descriptor.height {
 
-        // TODO(gw): Handle updates to size/format!
-        debug_assert_eq!(existing_item.allocated_rect.size.width, descriptor.width);
-        debug_assert_eq!(existing_item.allocated_rect.size.height, descriptor.height);
+            self.free_item_rect(existing_item.clone());
+
+            self.allocate_impl(
```

Comment:
this would allocate but not really fill it with data we got in the arguments (`data`), so I think `return` should be removed